testcontainers:1.17.1 and rest-assured:4.5.1 collision on hamcrest/hamcrest-core

### Describe the bug

Both
```
io.quarkus:quarkus-bom
io.quarkus.platform:quarkus-bom
```
bring in transitive dependencies:
```
+- org.testcontainers:testcontainers:jar:1.17.1:test
|  +- junit:junit:jar:4.13.2:test
|  |  \- org.hamcrest:hamcrest-core:jar:1.3:test
```
```
\- io.rest-assured:rest-assured:jar:4.5.1:test
   +- org.hamcrest:hamcrest:jar:2.1:test
```
These two may cause a collision, e.g. when `testcontainers` are on the classpath and `org.hamcrest.Matchers.containsStringIgnoringCase` is used in a test.
pom.xml:
```
...
    <dependency>
      <groupId>org.testcontainers</groupId>
      <artifactId>testcontainers</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.rest-assured</groupId>
      <artifactId>rest-assured</artifactId>
      <scope>test</scope>
    </dependency>
...
```
test:
```
package org.acme;

import static io.restassured.RestAssured.given;
import static org.hamcrest.Matchers.containsStringIgnoringCase;

import org.junit.jupiter.api.Test;

import io.quarkus.test.junit.QuarkusTest;

@QuarkusTest
public class GreetingResourceTest {

    @Test
    public void testHelloEndpoint() {
        given()
                .when().get("/hello")
                .then()
                .statusCode(200)
                .body(containsStringIgnoringCase("Hello from RESTEasy Reactive"));
    }

}
```
(See reproducer.)

In this simple case, a workaround is easy (reorder dependencies, exclude), but IMO the BOM should handle that instead.

### Expected behavior

No test errors.

### Actual behavior

```
[ERROR] org.acme.GreetingResourceTest.testHelloEndpoint  Time elapsed: 1.592 s  <<< ERROR!
java.lang.NoSuchMethodError: 'org.hamcrest.Matcher org.hamcrest.core.StringContains.containsStringIgnoringCase(java.lang.String)'
	at org.acme.GreetingResourceTest.testHelloEndpoint(GreetingResourceTest.java:19)
```

### How to Reproduce?

[hamcrest-dependency-collision-reproducer.zip](https://github.com/quarkusio/quarkus/files/8562960/hamcrest-dependency-collision-reproducer.zip)

```
unzip hamcrest-dependency-collision-reproducer.zip
cd ./hamcrest-dependency-collision-reproducer
./mvnw test
```


### Output of `uname -a` or `ver`

Linux 5.15.6-200.fc35.x86_64

### Output of `java -version`

openjdk version "11.0.14" 2022-01-18 OpenJDK Runtime Environment Temurin-11.0.14+9 (build 11.0.14+9) OpenJDK 64-Bit Server VM Temurin-11.0.14+9 (build 11.0.14+9, mixed mode)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.8.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.4

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -317,6 +317,17 @@
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
+            <dependency>
+                <groupId>org.testcontainers</groupId>
+                <artifactId>testcontainers</artifactId>
+                <version>${testcontainers.version}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>org.hamcrest</groupId>
+                        <artifactId>hamcrest-core</artifactId>
+                    </exclusion>
+                </exclusions>
+            </dependency>
 
             <!-- OpenTelemetry components, imported as a BOM -->
             <dependency>

